docs/annotations.md

```diff
@@ -0,0 +1,73 @@
+# Ingress Annotations
+
+This file defines a list of annotations which are supported by various Ingress controllers (both those based on the common ingress code, and alternative implementations).  The intention is to ensure the maximum amount of compatibility between different implementations.
+
+All annotations are assumed to be prefixed with `ingress.kubernetes.io/` except where otherwise specified. There is no attempt to record implementation-specific annotations using other prefixes.  (Traefik in particular defines several of its own annotations which are not described here, and does not seem to support any of the standard annotations.)
+
+Key:
+
+* `nginx`: the `kubernetes/ingress` nginx controller
+* `gce`: the `kubernetes/ingress` GCE controller
+* `traefik`: Traefik's built-in Ingress controller 
+* `haproxy`: Joao Morais' [haproxy controller](https://github.com/jcmoraisjr/haproxy-ingress)
+* `trafficserver`: Torchbox's [Apache Traffic Server controller plugin](https://github.com/torchbox/k8s-ts-ingress)
+
+## TLS-related
+
+| Name | Meaning
+| --- | ---
+| `ssl-passthrough` | Pass TLS connections directly to backend; do not offload.  Default `false`.  (nginx)
+| `ssl-redirect` | Redirect non-TLS requests to TLS when TLS is enabled.  Default `true`.  (nginx, trafficserver)
+| `force-ssl-redirect` | Redirect non-TLS requests to TLS even when TLS is not configured.  Default `false`.  (nginx, trafficserver).
+| `secure-backends` | Use TLS to communicate with origin (pods).  Default `false`. (nginx, trafficserver)
+| `kubernetes.io/ingress.allow-http` | Whether to accept non-TLS HTTP connections.  (gce)
+| `hsts-max-age` | Set an HSTS header with this lifetime. (trafficserver)
+| `hsts-include-subdomains` | Add includeSubdomains to the HSTS header. (trafficserver)
+
+## Authentication related
+
+| Name | Meaning
+| --- | ---
+| `auth-type` | Authentication type: `basic`, `digest`, ... (nginx, haproxy, trafficserver)
+| `auth-secret` | Secret name for authentication. (nginx, haproxy, trafficserver)
+| `auth-realm` | Authentication realm. (nginx, haproxy, trafficserver)
+| `auth-tls-secret` | Name of secret for TLS client certification validation. (nginx, haproxy)
+| `auth-tls-verify-depth` | Maximum chain length of TLS client certificate. (nginx)
+| `auth-satisfy` | Behaviour when more than one of `auth-type`, `auth-tls-secret` or `whitelist-source-range` are configured: `all` (default) or `any`. (trafficserver) | `trafficserver`
+| `whitelist-source-range` | Comma-separate list of IP addresses to restrict access to. (nginx, haproxy, trafficserver)
+
+## URL related
+
+| Name | Meaning
+| --- | ---
+| `app-root` | Redirect requests outside this path prefix to this prefix. (nginx, trafficserver)
```

Comment:
`app-root`, actually, "Configures the path to redirect to when the root path `/` of the domain is requested from the client" but with a better english then mine. haproxy also implements.